packages/app-builder/src/routes/ressources+/settings+/data-model+/tables+/$tableId+/exported-fields.ts

```diff
@@ -0,0 +1,150 @@
+import { isAdmin } from '@app-builder/models/user';
+import { initServerServices } from '@app-builder/services/init.server';
+import { getRoute } from '@app-builder/utils/routes';
+import { type ActionFunctionArgs, redirect } from '@remix-run/node';
+import z from 'zod';
+
+const ingestedDataFieldSchema = z.object({
+  path: z.array(z.string()),
+  name: z.string(),
+});
+
+const exportedFieldSchema = z.union([
+  z.object({ triggerObjectField: z.string() }),
+  z.object({ ingestedDataField: ingestedDataFieldSchema }),
+]);
+
+export async function action({ request, params }: ActionFunctionArgs) {
+  const { authService } = initServerServices(request);
+  const { user, dataModelRepository } = await authService.isAuthenticated(request, {
+    failureRedirect: getRoute('/sign-in'),
+  });
+
+  if (!isAdmin(user)) {
+    return redirect(getRoute('/'));
+  }
+
+  const { tableId } = params;
+  if (!tableId) {
+    return Response.json({ error: 'Table ID is required' }, { status: 400 });
+  }
+
+  // * POST * //
+
+  if (request.method === 'POST') {
+    const body = await request.json();
+
+    const parsedCreate = exportedFieldSchema.safeParse(body);
+    if (!parsedCreate.success) {
+      return Response.json({ error: 'Invalid payload' }, { status: 400 });
```

Comment:
Please look at other actions when creating yours, to keep constistency we always return a `{ success: boolean, ... }`  object. If some don't it will be harder to create a layer of abstraction of the browser side.